### DISCUSSION TITLE
To Rive or Not to Rive: Rive Animation & Elements

### DISCUSSION THREAD

**Discussion by aripuguh** [GECO-Project-App/facilitation-app#80] [How we work] [answered]
**Objective:** 
To find out whether it is beneficial to put the text (translation), and timer embedded in the Rive file(s). Shall we have all the elements on Rive or focus on the motion animation elements instead?

Findings:
Translation & Timer can be done by **setting up dynamic text for runtime** - https://www.youtube.com/watch?v=5kStqO626hk
Not by a State Machine mechanism. Then the dev team can tie it to the code.

**Text Runs:** https://www.youtube.com/watch?v=0OyB1YlDTfA&list=PLujDTZWVDSsFGonP9kzAnvryowW098-p3&index=41

We need to agree on the matter. **Practicality** wise? What's the **Pros & Cons**?

**Pros:**
- We can animate it here and there
- Looks more appealing

**Cons:**
- Efficiency?
- Naming agreement has to be met from the design & devs' sides at the front
- Do we need an Upgrade Voyager version or the free ones?
- Since we are mobile-first/oriented, not sure if we should do anything too polish. Ex: https://rive.app/community/files/8925-17057-demo-menu/
- Time-wise, workload consideration

**Block:** I can't really see it on runtime, so I can't tell about the result whether it is functioning/successful or not.

Feel free to add your pros & cons & thought here :) @alice-ang @eliasfaltin @Cristian-Nass

**Comment by eliasfaltin** [comment, marked as answer]
Update:
We discussed this during today's stand up and decided to focus on only doing animations in Rive for now, utilizing, at most, inputs to help Ari "step through" animations using the state machine (in, for example, the "pass it on method" animations).
